js/contextMenus.js

```diff
@@ -376,6 +378,15 @@ const siteSingleDetailTemplate = (siteKey, type, activeFrame) => {
     )
   }
 
+  if (paymentsEnabled && (type === siteTags.HISTORY || type === siteTags.BOOKMARK)) {
+    let location = siteDetail.get('location')
+    let enabled = ledgerUtil.shouldShowMenuOption(state, location)
+    template.push(
```

Comment:
I think that we should check if enabled and only if enabled we should add this template

Reply:
Template only added if menu option should be enabled.